compiler/utils.go

```diff
@@ -643,3 +644,38 @@ func endsWithReturn(stmts []ast.Stmt) bool {
 func encodeIdent(name string) string {
 	return strings.Replace(url.QueryEscape(name), "%", "$", -1)
 }
+
+// formatJSTag returns a string of Javascript code appropriate for accessing
+// the property identified by jsTag. If the jsTag value can be safely encoded
+// in Javascript using the dot notation this is used, else we fallback to the
+// bracket notation. For more details see:
+//
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
+//
+// Uses definition of an identifier from
+// https://developer.mozilla.org/en-US/docs/Glossary/Identifier
+func formatJSTag(jsTag string) string {
```

Comment:
I feel like the word "js tag" is confusing. Isn't this actually a struct tag for js? How about `formatStructTagForJS`? (I'm not an English native speaker so my suggestion might not sound natural)

Reply:
Addressed in the latest PR - I chose a slightly different version to keep more consistency with https://github.com/gopherjs/gopherjs/wiki/JavaScript-Tips-and-Gotchas